packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -257,7 +257,13 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
     }
 
     if (editCellState != null && column.renderEditCell) {
-      const params = { ...cellParams, ...editCellState, api: apiRef.current };
+      let updatedRow = row;
+      if (apiRef.current.unstable_getRowWithUpdatedValues) {
+        // Only the new editing API has this method
+        updatedRow = apiRef.current.unstable_getRowWithUpdatedValues(rowId, column.field);
+      }
```

Comment:
Is something blocking in the previous API to create the same method?

Reply:
Nothing blocks. I was considering that the legacy API wouldn't receive more fixes or improvements. Should it be added there too?